synapse/lib/auth.py

```diff
@@ -422,12 +422,20 @@ async def _setUserInfo(self, iden, name, valu, gateiden=None, logged=True, mesg=
             self.checkUserLimit()
 
         if gateiden is not None:
-            info = user.genGateInfo(gateiden)
-            info[name] = s_msgpack.deepcopy(valu)
             gate = self.reqAuthGate(gateiden)
-            gate.users.set(iden, info)
 
-            user.info['authgates'][gateiden] = info
+            admin = gate.gateusers.get(iden, {}).get('admin', False)
+
+            if name == 'rules' and not valu and not admin:
```

Comment:
Should we also pop the gate if they have no rules, and are going from admin=True to admin=False?